python-package/lightgbm/engine.py

```diff
@@ -341,7 +341,7 @@ def train(
                 )
         except callback.EarlyStopException as earlyStopException:
             booster.best_iteration = earlyStopException.best_iteration + 1
-            evaluation_result_list = earlyStopException.best_score
+            evaluation_result_list = [item if len(item) == 4 else item[:4] for item in earlyStopException.best_score]
```

Comment:
```suggestion
            # eval results from cv() have a 5th element with the standard deviation of metrics,
            # which is not needed for early stopping
            evaluation_result_list = [item[:4] for item in earlyStopException.best_score]
```

Since this can only have 4-element or 5-element tuples, I think this could be simplified to "always take the first 4 elements". Would that satisfy `mypy`?

It would also be good to add a comment here explaining why we're doing this... I think it'd be very difficult for someone who isn't familiar with `lightgbm`'s internals to understand why this is here.

Reply:
Also noting that hopefully this type of issue will go away soon as part of the work I'm doing #6748

Doesn't change my support for this PR, just mentioning that issue in case you hadn't seen it, so you know there are improvements coming to this "sometimes eval results are 4-element tuples, sometimes 5-element" situation.

Reply:
I've talked to @borchero offline and he said I can take over this PR he started. I'm going to apply this suggestion.